### PR TITLE
Update eslint-plugin-shopify v31.0.0

### PR DESCRIPTION
**Pull request by cartogram**
Related https://github.com/Shopify/eslint-plugin-shopify/pull/433

This PR upgrades this project to the latest `eslint` and `eslint-plugin-shopify`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,12 +32,9 @@
...
       "shopify/jest/no-snapshots": "off",
-      "shopify/jest/no-vague-titles": [
-        "error",
-        {
-          "allow": "all"
-        }
-      ]
+      "@typescript-eslint/ban-ts-ignore": "off",
+      "@typescript-eslint/no-unnecessary-condition": "off",
+      "@typescript-eslint/no-misused-promises": "off"
```
Comment (lemonmade) [inline package.json]: Were all of these so pervasive they needed to be disabled?

File: packages/graphql-typescript-definitions/package.json [hunk 2]
```diff
@@ -43,7 +43,10 @@
     "glob": "^7.1.2",
     "graphql": "^14.0.0",
     "graphql-config-utilities": "^1.0.0",
-    "graphql-tool-utilities": "^1.1.0"
+    "graphql-tool-utilities": "^1.1.0",
+    "yargs":"^14.2.0",
+    "@types/yargs": "11.0.0",
+    "graphql-config": "^2.2.1"
```
Comment (lemonmade) [inline packages/graphql-typescript-definitions/package.json]: Let's do `^2.0.0` here instead

File: packages/graphql-typescript-definitions/package.json [hunk 3]
```diff
@@ -43,7 +43,10 @@
     "glob": "^7.1.2",
     "graphql": "^14.0.0",
     "graphql-config-utilities": "^1.0.0",
-    "graphql-tool-utilities": "^1.1.0"
+    "graphql-tool-utilities": "^1.1.0",
+    "yargs":"^14.2.0",
```
Comment (lemonmade) [inline packages/graphql-typescript-definitions/package.json]: Let's do `^14.0.0` here instead

File: packages/graphql-typescript-definitions/src/index.ts [hunk 4]
```diff
@@ -99,32 +99,34 @@ export class Builder extends EventEmitter {
...
   once(event: 'build:docs', handler: (built: DocumentBuild) => void): this;
   once(event: 'build:schema', handler: (built: SchemaBuild) => void): this;
-  once(event: 'start:docs', handler: () => void): this;
-  once(event: 'end:docs', handler: () => void): this;
-  once(event: 'start:schema', handler: () => void): this;
-  once(event: 'end:schema', handler: () => void): this;
+  once(
+    event: 'start:docs' | 'end:docs' | 'start:schema' | 'end:schema',
+    handler: () => void,
+  ): this;
```
Comment (lemonmade) [inline packages/graphql-typescript-definitions/src/index.ts]: Neat, a rule does this too?

File: packages/graphql-validate-fixtures/package.json [hunk 5]
```diff
@@ -34,6 +34,7 @@
     "graphql": "^14.0.0",
     "graphql-config-utilities": "^1.0.0",
     "graphql-tool-utilities": "^1.1.0",
-    "yargs": "^11.0.0"
+    "yargs": "^11.0.0",
+    "graphql-config": "^2.2.1"
```
Comment (lemonmade) [inline packages/graphql-validate-fixtures/package.json]: Let's do `^2.0.0` here instead

File: packages/graphql-typescript-definitions/src/index.ts (hunk 4 above)
Comment (cartogram) [reply packages/graphql-typescript-definitions/src/index.ts]: I know right? https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/unified-signatures.md

File: package.json (hunk 1 above)
Comment (cartogram) [reply package.json]: Yeah, I'll add a comment to mention that.

File: package.json [hunk 6]
```diff
@@ -66,8 +64,8 @@
     "@types/jest": "^24.0.11",
     "@types/node": "^10.0.4",
     "@types/yargs": "^11.0.0",
-    "eslint": "^5.16.0",
-    "eslint-plugin-shopify": "^30.0.1",
+    "eslint": "^6.5.0",
+    "eslint-plugin-shopify": "31.0.0",
```
Comment (patsissons) [inline package.json]: is the strict version intentional?

**Pull request merged**